addons/event_sale/models/sale_order.py

```diff
@@ -21,7 +21,12 @@ def write(self, vals):
         return result
 
     def action_confirm(self):
+        to_open_registration = self.order_line.registration_ids.filtered(
+            lambda reg: reg.state in ["draft", "cancel"]
+        )
         res = super(SaleOrder, self).action_confirm()
+        to_open_registration._update_mail_schedulers()
```

Comment:
Maybe we should be defensive ?
```suggestion
        to_open_registration.filtered(lambda r: r.state == "confirm").._update_mail_schedulers()
```

Reply:
If you mean "open" instead of "confirm" (working on registration state, not sale order) then I removed it since the `_update_mail_schedulers` already do the `filtered`, is it ok?